### ISSUE TITLE
golang.org/x/crypto/bn256:  wrong formulas

### ISSUE THREAD

**Issue by zhangzhanli** [golang/go#48143]
bn256/gfp6.go : 251 in latest version of go

// On the other hand (xj²τ² + yjτ + z)(xjτ² + yj²τ + z)
// = τ²(y²-ξxz) + τ(ξx²-yz) + (z²-ξxy)
//
// So that's why A = (z²-ξxy), B = (ξx²-yz), C = (y²-ξxz)

while the right one is :

//On the other hand (xj²τ² + yjτ + z)(xjτ² + yj²τ + z)
//= τ²(y²-xz) + τ(ξx²-yz) + (z²-ξxy)
//
//So that's why A = (z²-ξxy), B = (ξx²-yz), C = (y²-xz)

in addition,  http://arxiv.org/pdf/0904.0854v3.pdf which is referenced by optate.go : 9  also has wrong formulas :

<img width="1000" alt="截屏2021-09-02 下午5 42 01" src="https://user-images.githubusercontent.com/32356175/131823681-4d693367-86ce-4eef-b090-a2f6348edb16.png">

**Comment by cherrymui** [comment]
It looks the code actually does
```
        C := newGFp2(pool)
        C.Square(a.y, pool)
        t1.Mul(a.x, a.z, pool)
        C.Sub(C, t1)
```
i.e. `C = (y²-xz)` without ξ.

@zhangzhanli do you suggest only the comment is wrong or the implementation as well? Thanks.

cc @FiloSottile @agl @katiehockman @rolandshoemaker

**Comment by zhangzhanli** [comment]
@cherrymui  Yes, it seems the source code works as expected.  As for full addition, since we use mixed addition instead , so there is no impact on miller's algorithm.

**Comment by cherrymui** [comment]
Thanks.
